tools/super-tox.py

```diff
@@ -69,32 +99,33 @@ def _patch_requirements_file(
                 continue
             if req.name != "ops":
                 adjusted.append(line)
-    if ops_source_branch:
-        adjusted.append(f"\ngit+{ops_source}@{ops_source_branch}\n")
+    if settings.ops_source_branch:
+        adjusted.append(f"\ngit+{settings.ops_source}@{settings.ops_source_branch}\n")
     else:
-        adjusted.append(f"\ngit+{ops_source}\n")
+        adjusted.append(f"\ngit+{settings.ops_source}\n")
     with requirements.open("w") as req:
         req.write("\n".join(adjusted))
     return "".join(original)
 
 
 @contextlib.contextmanager
-def patch_ops(location: pathlib.Path, ops_source: str, ops_source_branch: str | None):
+def patch_ops(location: pathlib.Path):
     requirements = location / "requirements.txt"
     pyproject = location / "pyproject.toml"
     if requirements.exists():
-        original = _patch_requirements_file(requirements, ops_source, ops_source_branch)
+        original = _patch_requirements_file(requirements)
         # Annoyingly, sometimes there's also a requirements-*.txt file that also
         # has an ops dependency, so patch those as well.
         # We could perhaps parse the tox.ini file to find the appropriate
         # dependency declarations matching the specified environment (or all
         # environments, if one isn't), but this seems sufficient for now.
         extras = {}
-        for fn in requirements.parent.glob("requirements-*.txt"):
-            extras[fn] = _patch_requirements_file(fn, ops_source, ops_source_branch)
-        # Also, in some cases, requirements*.in files.
-        for fn in requirements.parent.glob("requirements*.in"):
-            extras[fn] = _patch_requirements_file(fn, ops_source, ops_source_branch)
+        for fn in itertools.chain(
+            requirements.glob("requirements-*.txt"),  # e.g. requirements-unit.txt
+            requirements.glob("*-requirements.txt"),  # e.g. test-requirements.txt
```

Comment:
```suggestion
            requirements.glob("*requirements*.txt"),  # e.g. requirements-unit.txt, test-requirements.txt
```

What do you think? Too broad and should just stick with the two more explicit ones?

Reply:
I wonder, here are some weird ones:

```
.cache/training-operator/requirements-fmt.in
.cache/training-operator/requirements-fmt.txt
.cache/training-operator/requirements-integration.in
.cache/training-operator/requirements-integration.txt
.cache/training-operator/requirements-lint.in
.cache/training-operator/requirements-lint.txt
.cache/training-operator/requirements-unit.in
.cache/training-operator/requirements-unit.txt
.cache/training-operator/requirements.in
.cache/training-operator/requirements.txt
```

And

```
.cache/wordpress-k8s-operator/tests/integration/requirements.txt
.cache/wordpress-k8s-operator/tests/integration/requirements_juju3.txt
```


Besides these, `test-/-test` and `dev-/-dev` are rather common.

Reply:
full list

```command
> fd requirements .cache/ | sed -e 's|.*/||' | sort | uniq -c | sort -n
   1 build-requirements.txt
   1 cookie-requirements.txt
   1 requirements-charmcraft.txt
   1 requirements-integration-v1.in
   1 requirements-integration-v1.txt
   1 requirements-integration-v2.in
   1 requirements-integration-v2.txt
   1 requirements-lxml.txt
   1 requirements-test.txt
   1 requirements_juju3.txt
   3 r-system-requirements.md
   8 fmt-requirements.txt
   8 integration-requirements.txt
   8 lint-requirements.txt
   8 r-requirements.md
   8 unit-requirements.txt
  11 requirements-dev.txt
  20 test-requirements.in
  22 test-requirements.txt
  37 requirements-unit.in
  37 requirements-unit.txt
  38 requirements-integration.in
  38 requirements-integration.txt
  43 requirements-fmt.in
  43 requirements-fmt.txt
  43 requirements-lint.in
  44 requirements-lint.txt
  60 requirements.in
 261 requirements.txt
```

Reply:
I'm fine either way, also you're the maintainer you can insist on something :)

Reply:
Let's leave it as-is. The intention is that some standardisation happens here, although I'm not sure how quickly that will happen.

I wonder if super-tox ought to be trying to do this in a smarter way, like if there's a poetry lock file then just run the poetry commands to change the dependency, same for uv. And then falling back for the other cases. But something for another time.